### PR TITLE
[runtime] Don't call ExecutionEngine constructor in aot mode

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/mini/mini.c [hunk 1]
```diff
@@ -4194,8 +4194,8 @@ mono_jit_compile_method_inner (MonoMethod *method, MonoDomain *target_domain, in
 		char *fullname = mono_method_full_name (method, TRUE);
 		char *msg = g_strdup_printf ("Attempting to JIT compile method '%s' while running with --aot-only. See http://docs.xamarin.com/ios/about/limitations for more information.\n", fullname);
 
-		ex = mono_get_exception_execution_engine (msg);
-		mono_error_set_exception_instance (error, ex);
+		mono_error_set_execution_engine (error, msg);
```
Comment (lambdageek) [inline mono/mini/mini.c]: `mono_error_set_*` will do a strdup internally, so you can just write `mono_error_set_execution_engine (error, "Attempting to ...", fullname)`

Comment (alexanderkyte) [inline mono/mini/mini.c]: Fixed

**Comment by lambdageek** [comment]
LGTM.

@monojenkins merge

**Comment by vargaz** [comment]
The failures are unrelated.

**Pull request merged**